modular_skyrat/modules/powerarmor/code/powerarmor.dm

```diff
@@ -0,0 +1,839 @@
+	//this means that after putting in this part, you must do these actions before continuing
+	var/list/powerarmor_tool = list(0)
+
+/obj/item/powerarmor/powerarmor_part/head
+	name = "head power armor part"
+	icon_state = "head"
+	powerarmor_part = 1
+	powerarmor_tool = list(3)
+
+/obj/item/powerarmor/powerarmor_part/chest
+	name = "chest power armor part"
+	icon_state = "chest"
+	powerarmor_part = 2
+	powerarmor_tool = list(2,3)
+
+/obj/item/powerarmor/powerarmor_part/larm
+	name = "left arm power armor part"
+	icon_state = "larm"
+	powerarmor_part = 3
+	powerarmor_tool = list(1)
+
+/obj/item/powerarmor/powerarmor_part/rarm
+	name = "right arm power armor part"
+	icon_state = "rarm"
+	powerarmor_part = 4
+	powerarmor_tool = list(1)
+
+/obj/item/powerarmor/powerarmor_part/lleg
+	name = "left leg power armor part"
+	icon_state = "lleg"
+	powerarmor_part = 5
+	powerarmor_tool = list(1)
+
+/obj/item/powerarmor/powerarmor_part/rleg
+	name = "right leg power armor part"
+	icon_state = "rleg"
+	powerarmor_part = 6
+	powerarmor_tool = list(1)
+
+/obj/item/clothing/suit/hooded/powerarmor
+	name = "power armor suit"
+	desc = "A suit for the power armor that is capable of being modified to give the user even greater potential."
+	icon = 'modular_skyrat/modules/powerarmor/icons/suits.dmi'
+	worn_icon = 'modular_skyrat/modules/powerarmor/icons/suit.dmi'
+	icon_state = "powersuit"
+	hoodtype = /obj/item/clothing/head/hooded/powerarmor
+	body_parts_covered = HEAD|CHEST|GROIN|LEGS|ARMS
+	equip_delay_self = 50
+	strip_delay = 50
+	slowdown = 0.5
+
+	light_power = 0.75
+
+	mutant_variants = NONE
+	flags_inv = HIDEGLOVES|HIDESHOES|HIDEJUMPSUIT|HIDESEXTOY
+
+	///the maximum amount of upgrades it can have (where some upgrades can cost multiple)
+	var/upgradelimit = 20
+	///whether the upgradelimit has been up'd, by some item
+	var/upgradeboosted = FALSE
+	//the cooldown for
+	COOLDOWN_DECLARE(healing_cooldown)
+	//the list of upgrades possible, starting here
+	var/list/armor_upgraded = list(0, 0, 0, 0, 0, 0, 0) //melee, bullet, laser, energy, bio, rad, fire
+	var/list/healing_upgraded = list(FALSE, FALSE, FALSE, FALSE, FALSE) //brute, burn, toxin, oxygen, stamina
+	var/speed_upgraded = 0
+	var/list/misc_upgraded = list(FALSE, FALSE, FALSE, FALSE, FALSE) //spaceproof, light, welding, temp-regulating, storage
+
+	var/mob/wearer
+	var/obj/item/storage/backpack/inner_backpack
+
+/obj/item/clothing/suit/hooded/powerarmor/emp_act(severity)
+	. = ..()
+	if(wearer && isliving(wearer))
+		var/mob/living/living_wearer = wearer
+		living_wearer.Stun(5 SECONDS)
+		living_wearer.adjustFireLoss(25)
+		to_chat(living_wearer, span_warning("[src] short-circuits, hurting you in the process!"))
+
+/obj/item/clothing/suit/hooded/powerarmor/AltClick(mob/user)
+	. = ..()
+	var/list/selection = list()
+	if(misc_upgraded[5])
+		selection += "storage"
+	var/get_choice = tgui_input_list(user, "Choose which option to selection", "Selection Menu", selection)
```

Comment:
but theres only 1 choice youll ever get, storage

Reply:
open to allow expansion.